### PR TITLE
[ENH] Group shading

### PR DESCRIPTION
**Pull request by ryanhammonds**
This is a point in #193 and adds shading around group spectra. By default, the shading defaults to +/- 1 stdev. This is bypassed if a 1d array is passed to `shade`.

```python
import numpy as np
from neurodsp.sim import sim_combined
from neurodsp.spectral import compute_spectrum
from fooof.plts import plot_spectra
from fooof.plts.error import plot_error_shade

components = {'sim_bursty_oscillation': {'freq': 10},
              'sim_powerlaw': {'exponent': -2}}

n_iters = 20

powers = np.zeros((n_iters, 50))

for idx in range(n_iters):
    sig = sim_combined(10, 1000, components)
    freqs, power = compute_spectrum(sig, 1000, f_range=(1, 50))
    powers[idx] = power
    
plot_spectra(freqs, powers, log_powers=True)    
plot_error_shade(freqs, powers, log_powers=True)
```
shade

### CODE REVIEW

**Comment by TomDonoghue** [comment]
I updated this func with some of the changes we chatted about recently including moving, renaming (though I'm still not sure about the name), and updating approaches for calculate average and shade. @ryanhammonds - can you check the updates, and see if this looks good to you, and test it out that everything works? Thanks!

**Review by ryanhammonds — COMMENTED** [review]
I left a few comments that I'm going to push updates for. If you agree with the comments/changes then this should be ready to merge. The changes were related to getting custom callables working:
```python
def average_callable(powers):
    
    return np.mean(powers, axis=0)

def shade_callable(powers):
    
    return np.std(powers, axis=0)

plot_spectra_yshade(freqs, powers, shade=shade_callable,
                    average=average_callable, log_powers=True)
```

File: fooof/plts/spectra.py [hunk 1]
```diff
@@ -172,3 +173,70 @@ def plot_spectra_shading(freqs, power_spectra, shades, shade_colors='r',
...
+    avg_powers = avg_func(plt_powers, axis=0) if plt_powers.ndim == 2 else plt_powers
+
+    # Plot average power spectrum
+    ax.plot(plt_freqs, avg_powers, linewidth=2.0, color=color, label=label)
+
+    # Organize shading to plot
+    shade_funcs = {'std' : np.std, 'sem' : sem}
+    shade_func = shade_funcs[shade] if isinstance(shade, str) else shade
+    shade_vals = scale * shade_func(plt_powers, axis=0) \
+        if isinstance(shade, str) else scale * shade
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply fooof/plts/spectra.py]: lol, ooops.

File: fooof/plts/spectra.py [hunk 2]
```diff
@@ -172,3 +173,70 @@ def plot_spectra_shading(freqs, power_spectra, shades, shade_colors='r',
...
+    ax = check_ax(ax, plot_kwargs.pop('figsize', PLT_FIGSIZES['spectral']))
+
+    # Set plot data & labels, logging if requested
+    plt_freqs = np.log10(freqs) if log_freqs else freqs
+    plt_powers = np.log10(power_spectra) if log_powers else power_spectra
+
+    # Organize mean spectrum to plot
+    avg_funcs = {'mean' : np.mean, 'median' : np.median}
+    avg_func = avg_funcs[average] if isinstance(average, str) else average
+    avg_powers = avg_func(plt_powers, axis=0) if plt_powers.ndim == 2 else plt_powers
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply fooof/plts/spectra.py]: Yeh, great points. I think what you've done makes a ton of sense to address this.

**Comment by TomDonoghue** [comment]
@ryanhammonds - thanks for the review here! You are totally correct, I totally dropped the ball on actually implementing callable support lol (oops - too much late night coding). I like your updates here - merging this in.

**Pull request merged**